app/models/concerns/folio/auto_translate_for_traco.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require "deepl"
+
+module Folio::AutoTranslateForTraco
+  extend ActiveSupport::Concern
+
+  included do
+    DeepL.configure do |config|
+      config.auth_key = ENV.fetch("DEEPL_API_KEY", nil)
+      config.host = ENV.fetch("DEEPL_API_HOST", "https://api-free.deepl.com")
+    end
+  end
+
+  class_methods do
+    def auto_translates(*attributes)
+      attributes.each do |attribute|
+        define_method("#{attribute}=") do |val|
+          I18n.available_locales.each do |l|
+            return if send("#{attribute}_#{l}").present?
+
+            translated = ::DeepL.translate(val, I18n.locale, l).text
+            send("#{attribute}_#{l}=", translated)
+
+          rescue ::DeepL::Exceptions::RequestError, ::DeepL::Exceptions::NotSupported, DeepL::Exceptions::Error => e
+            Rails.logger.error("DeepL error: #{e.message}")
+            send("#{attribute}_#{l}=", val)
+          end
+
+          translate_slugs_if_new_record
```

Comment:
@jirkamotejl proc se tohle vola v kazdem setteru? Nemelo by to byt jednou v `before_validation`?